crates/ironrdp-connector/src/connection.rs

```diff
@@ -53,6 +56,173 @@ pub struct ConnectionResult {
     pub no_server_pointer: bool,
 }
 
+// FIXME: move that to a new module at ironrdp_connector::credssp
+
+pub type CredSspProcessGenerator<'a> = Generator<'a, NetworkRequest, sspi::Result<Vec<u8>>, sspi::Result<ClientState>>;
+
+#[derive(Debug)]
+pub struct CredSspSequence {
+    client: CredSspClient,
+    next_request: Option<credssp::TsRequest>,
+    state: CredSSPState,
+    selected_protocol: nego::SecurityProtocol,
+}
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum CredSSPState {
+    CredsspInitial,
+    CredsspReplyNeeded,
+    CredsspEarlyUserAuthResult,
```

Comment:
Need Help to Test this state

Reply:
I’ll try to run the code on my side and see if I can hit it